test/integration_test/migration_test.go

```diff
@@ -55,22 +68,30 @@ func triggerMigrationTest(
 		require.NoError(t, err, "Error resetting remote config")
 	}()
 
-	err = schedulerDriver.WaitForRunning(ctxs[0], defaultWaitTimeout, defaultWaitInterval)
+	timeout := defaultWaitTimeout
+	if !migrationSuccessExpected {
+		timeout = timeout / 2
+	}
+	err = schedulerDriver.WaitForRunning(ctxs[0], timeout, defaultWaitInterval)
 	if migrationSuccessExpected {
 		require.NoError(t, err, "Error waiting for migration to get to Ready state")
 
-		// wait on cluster 2 to get mysql pod running
+		// wait on cluster 2 for the app to be running
 		err = setRemoteConfig(remoteFilePath)
 		require.NoError(t, err, "Error setting remote config")
 		err = schedulerDriver.WaitForRunning(preMigrationCtx, defaultWaitTimeout, defaultWaitInterval)
 		require.NoError(t, err, "Error waiting for pod to get to running state on remote cluster after migration")
-		// destroy mysql app on cluster 2
+		if !migrateAllAppsExpected {
+			err = schedulerDriver.WaitForRunning(allAppsCtx, defaultWaitTimeout/2, defaultWaitInterval)
```

Comment:
This assumes that `WaitForRunning` will fail because of absence of the apps which shouldn't be migrated. But we could have a case where the apps are migrated but are crashing. In that case, the test will not catch the migration issue.

We should rather check specifically for absence of the app which doesn't match the label selector. If that's too specific, we can even parse the returned err for a not found error.

Reply:
Yeah torpedo isn't flexible enough to do that. I'm actually going to use sched-ops package directly in these tests to get more flexibility. But that is future work.